### PR TITLE
docker-fixed Kernel upgrade for Ubuntu/Centos. mesos-agent role replaced

### PR DESCRIPTION
**Pull request by dvictor1**
role mesos-slave is more convenient
Centos kernel deploy reqd improvement using yumrepo ansible module

### CODE REVIEW

**Comment by jonas** [comment]
I suggest that either PRs are smaller or are split in multiple commits for each change.

Regarding the slave/agent I used agent because Mesos will start using agent from version 1.0 (see https://issues.apache.org/jira/browse/MESOS-1478). I am OK changing it if you think it makes it easier to understand what is going on.

File: vagrant/hosts [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 [all:vars]
 ansible_ssh_user=vagrant
-ansible_dns=mesos.dev
+ansible_domainname=mesos.dev
```
Comment (jonas) [inline vagrant/hosts:7]: Yes, I think this is a better name. And I would even go further and suggest we completely avoid the `ansible_` namespace and use `mesos_domain`.

File: roles/mesos/tasks/RedHat.yml [hunk 2]
```diff
@@ -11,16 +11,18 @@
 # match those of the roles.
 
 - name: Check Mesos Master running
-  shell: sudo status mesos-master
```
Comment (jonas) [inline roles/mesos/tasks/RedHat.yml:14]: The `Debian.yml` file contains the following. Maybe we can move these lines to `main.yml`?

``` yaml
# The Mesos package installs both the master and agent. Make the Mesos services
# match those of the roles.
- name: Disable Mesos Master for non-master hosts
  service: name=mesos-master state=stopped enabled=no
  when: "'mesos-master' not in group_names"

- name: Disable Mesos Agent for non-agent hosts
  service: name=mesos-slave state=stopped enabled=no
  when: "'mesos-agent' not in group_names"
```

File: roles/common/templates/etc-hosts.j2 [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 # Mesos mappings
 {% for item in groups.all %}
-{{item}} {{hostvars[item]['ansible_hostname']}}.{{ansible_dns}} {{hostvars[item]['ansible_hostname']}}
+{{item}} {{hostvars[item]['mesos_hostname']}} {{hostvars[item]['ansible_hostname']}}
```
Comment (jonas) [inline roles/common/templates/etc-hosts.j2:6]: :+1:

File: vagrant/hosts (hunk 1 above)
Comment (dvictor1) [inline vagrant/hosts:7]: And you are right, ansible 2.1 puts wrong data into files, when using ansible_dns, I had to change the name.

File: roles/mesos/tasks/RedHat.yml (hunk 2 above)
Comment (jonas) [inline roles/mesos/tasks/RedHat.yml:14]: @dvictor1 Why did the Ansible `service` task not work on RedHat?

Comment (dvictor1) [inline roles/mesos/tasks/RedHat.yml:14]: service works only with /etc/init.d

**Comment by jonas** [comment]
@dvictor1 Let's get it merged

**Pull request merged**